src/engines/engine-decorator-manager.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+import {MultiMap} from '../utils';
+
+/**
+ * Engine decorator manager for plugins.
+ * @class EngineDecoratorManager
+ */
+class EngineDecoratorManager {
+  _decoratorProviders: MultiMap<string, IEngineDecoratorProvider> = new MultiMap();
```

Comment:
this can be a simple map, multi map is were the value is array but we only have one decorator per plugin